homeassistant/components/modbus/validators.py

```diff
@@ -130,21 +150,40 @@ def struct_validator(config: dict[str, Any]) -> dict[str, Any]:
         if not swap_type_validator:
             error = f"{name}: `{CONF_SWAP}:{swap_type}` cannot be combined with `{CONF_DATA_TYPE}: {data_type}`"
             raise vol.Invalid(error)
+    if (
+        register_size_bytes != RegisterBytes.NOT_SET
+        and not validator.allow_register_size_set
+    ):
+        error = f"{name}: `{CONF_REGISTER_SIZE_BYTES}:{register_size_bytes}` cannot be combined with `{CONF_DATA_TYPE}: {data_type}`"
+        raise vol.Invalid(error)
+    if register_size_bytes == RegisterBytes.NOT_SET:
+        if data_type in (DataType.INT8, DataType.UINT8):
```

Comment:
Please do not add UINT8, it does not work, neither do e.g. size=3

Reply:
I didnt add UINT8 or INT8 it already exists

Reply:
But you made it legal to use !!!